### PR TITLE
revert: Temporarily revert some breaking changes (for 2.0.0-internal.6.x)

### PR DESCRIPTION
**Pull request by alexvy86**
## Description

Additional reverts to ease transition to `2.0.0-internal.6.x`.

- Restore `IDisposable` and `EventForwarder` members on `PureDataObject` (partial revert of https://github.com/microsoft/FluidFramework/pull/16283).
- Restore `id` property on `IContainerContext` (partial revert of https://github.com/microsoft/FluidFramework/pull/16262)
- ~Restore `parent` property on `ISegment` (partial revert of https://github.com/microsoft/FluidFramework/pull/15985)~ trying to address this in a different way, so no reversal here in the end.

All of these reverts will be undone (i.e. changes re-applied) in `2.0.0-internal.7.0`.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

@ChumpChief , @anthony-murphy , would appreciate a check on your respective changes. I think the reverts for Matt's seem fine, but not so confident on Tony's, and reverting the whole PR felt like it might be overkill. But I couldn't quite track what was setting that `ISegment.parent`.

### CODE REVIEW

File: .changeset/eighty-weeks-relate.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+"@fluidframework/aqueduct": minor
+---
+
+PureDataObject temporarily extends EventForwarder and implements IDisposable again
+
+`PureDataObject` extends `EventForwarder` and implements `IDìsposable` again to ease the transition to `2.0.0-internal.6.x`.
+These changes will be undone in `2.0.0-internal.7.0.0`.
```
Comment (tylerbutler) [inline .changeset/eighty-weeks-relate.md:8]:
```suggestion
These interfaces will no longer be implemented on `PureDataObject` in version `2.0.0-internal.7.0.0`.
```

File: packages/framework/aqueduct/src/data-objects/pureDataObject.ts [hunk 2]
```diff
@@ -175,4 +192,42 @@ export abstract class PureDataObject<I extends DataObjectTypes = DataObjectTypes
...
+	/**
+	 * Called when the host container closes and disposes itself
+	 * @deprecated - 2.0.0-internal.5.2.0 - Dispose does nothing and will be removed in an upcoming release.
+	 */
+	public dispose(): void {
+		super.dispose();
+	}
+
+	/**
+	 * @deprecated - 2.0.0-internal.5.2.0 - PureDataObject does not actually set up to forward events, and will not be an EventForwarder
```
Comment (tylerbutler) [inline packages/framework/aqueduct/src/data-objects/pureDataObject.ts:205]:
```suggestion
	 * @deprecated 2.0.0-internal.5.2.0 - PureDataObject does not actually set up to forward events, and will not be an EventForwarder
```

no dashes after \@deprecated in TSDoc

File: api-report/merge-tree.api.md [hunk 3]
```diff
@@ -405,6 +405,10 @@ export interface IMergeNodeCommon {
     // (undocumented)
     isLeaf(): this is ISegment;
     ordinal: string;
+    // Warning: (ae-forgotten-export) The symbol "IMergeBlock" needs to be exported by the entry point index.d.ts
+    //
+    // @internal @deprecated (undocumented)
+    parent?: IMergeBlock;
```
Comment (tylerbutler) [inline api-report/merge-tree.api.md:411]: Probably need to export this.

File: packages/common/container-definitions/src/runtime.ts [hunk 4]
```diff
@@ -197,6 +197,17 @@ export interface IContainerContext {
 	updateDirtyContainerState(dirty: boolean): void;
 
 	readonly supportedFeatures?: ReadonlyMap<string, unknown>;
+
+	/**
+	 * WARNING: this id is meant for telemetry usages ONLY, not recommended for other consumption
+	 * This id is not supposed to be exposed anywhere else. It is dependant on usage or drivers
+	 * and scenarios which can change in the future.
+	 * @deprecated - 2.0.0-internal.5.2.0 - The docId is already logged by the IContainerContext.taggedLogger for
```
Comment (Josmithr) [inline packages/common/container-definitions/src/runtime.ts:205]:
```suggestion
	 * @deprecated 2.0.0-internal.5.2.0 - The docId is already logged by the IContainerContext.taggedLogger for
```

`@deprecated` blocks don't want a `-` separator. See: https://github.com/microsoft/FluidFramework/wiki/TSDoc-Guidelines#deprecated

I would also recommend using an `{@link}` tag where we reference `IContainerContext.taggedLogger` for the generated docs.

File: packages/framework/aqueduct/src/data-objects/pureDataObject.ts [hunk 5]
```diff
@@ -53,6 +56,14 @@ export abstract class PureDataObject<I extends DataObjectTypes = DataObjectTypes
 
 	protected initializeP: Promise<void> | undefined;
 
+	/**
+	 * @deprecated - 2.0.0-internal.5.2.0 - PureDataObject does not provide a functioning built-in disposed flow.
```
Comment (Josmithr) [inline packages/framework/aqueduct/src/data-objects/pureDataObject.ts:60]:
```suggestion
	 * @deprecated 2.0.0-internal.5.2.0 - PureDataObject does not provide a functioning built-in disposed flow.
```

**Review by Josmithr — COMMENTED** [review]
The changes seem reasonable to me, and the surrounding documentation seems clear enough, but I don't have a ton of context into what is / is not required to be reverted here.

I did leave a comment around the use of `@deprecated` tags though.

**Review by Josmithr — APPROVED** [review]

**Pull request merged**